sidecar/test_sidecar.py

```
@@ -0,0 +1,246 @@
+    ExtractResponse,
+    HealthResponse,
+    LABELS,
+    MAX_TEXT_CHARS,
+)
+from runtime import predict_windowed, calculate_iou
+
+
+class MockGlinerModel:
+    """Mock GLiNER model for deterministic unit tests without loading neural weights."""
+
+    def __init__(self, mock_responses: Dict[str, List[Dict[str, Any]]] = None):
```

Ruff (0.16.2)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)